keps/sig-instrumentation/4827-component-statusz/README.md

```diff
@@ -0,0 +1,422 @@
+
+### User Stories
+
+#### As a developer
+
+- I want to quickly identify the exact version of a binary running in production so I can correlate it with known issues or recent code changes
+- I want to quickly access the compatibility version information of a deployed binary so I can efficiently debug upgrade failures and identify potential compatibility issues with dependencies
+
+#### As an operator
+
+- I need to troubleshoot a performance issue and want to easily check for dependency conflicts or version mismatches without sifting through logs
+
+#### As support engineer
+
+- I want to access detailed build information to determine if a customer is running a known problematic version of a component
+
+### Risks and Mitigations
+
+1. **No sensitive data exposed**
+    
+  We will apply the  principle of least privilege by granting only the minimum necessary permissions for accessing /statusz, which is primarily a debugging tool for operators and maintainers.
+  /statusz will not be accessible through general discovery mechanisms like system:discovery. This prevents unintended exposure of potentially sensitive debugging information.
+  
+  We'll leverage Kubernetes RBAC to grant access to /statusz only to specific users and service accounts that require it for operational monitoring and debugging. For this, we will use the existing system:monitoring group ([ref](https://github.com/kubernetes/kubernetes/blob/release-1.31/staging/src/k8s.io/apiserver/pkg/authentication/user/user.go#L73)), ensuring that only authorized personnel can access this endpoint.
+  
+1. **Performance impact**
+    
+  We will take care to carefully consider the computational overhead associated with the statusz endpoint to avoid negatively impacting the server's overall performance.
+
+2. **Essential data only**
+    
+  We will prioritize inclusion of only essential diagnostic and monitoring data that aligns with the intended purpose of the z-page.
+
+3. **Premature Dependency on Unstable Format**
+
+  The alpha release will explicitly support plain text format only, making it clear that the output is not intended for parsing or automated monitoring. The feature will be secured behind a feature gate that is disabled by default, ensuring users opt-in consciously. Also, the endpoint will be implemented with versioned URLs to facilitate future transitions to structured formats without breaking existing integrations, once the usage patterns and requirements are better understood
+
+## Design Details
+
+### Data Format and versioning
+
+Initially, the statusz page will exclusively support a plain text format for responses. We will implement these endpoints using versioned URLs, like /statusz/v1, to ensure future compatibility. This versioning strategy allows us to seamlessly introduce structured data formats (e.g., JSON) in the future, through a distinct endpoint such as /statusz/v2, without disrupting existing implementations.
```

Comment:
Isn't the content-type header the right mechanism for different data-formats of the same basic content?

e.g. only support `application/json` for now and do more later.

That reserves "v2" for major changes

Reply:
Using the content-type header does make sense, can have `Content-Type: text/plain` for starters and modify that later on. Would still need versioned URLs to be able to distinguish the response schema though